exporters/otlp/recordable_test.cc

```diff
@@ -0,0 +1,64 @@
+#include "exporters/otlp/recordable.h"
+
+#include <gtest/gtest.h>
+
+using namespace opentelemetry::exporter::otlp;
+
+TEST(Recordable, SetIds)
+{
+    Recordable rec;
+
+    const uint8_t trace_id_buff[16] = {0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1};
+    const uint8_t span_id_buff[8] = {0,0,0,0,0,0,0,2};
+    const uint8_t parent_span_id_buff[8] = {0,0,0,0,0,0,0,3};
+
+    opentelemetry::nostd::span<const uint8_t, 16>trace_id_span(trace_id_buff, 16);
+    opentelemetry::nostd::span<const uint8_t, 8> span_id_span(span_id_buff, 8);
+    opentelemetry::nostd::span<const uint8_t, 8> parent_span_id_span(parent_span_id_buff, 8);
+
+    opentelemetry::trace::TraceId trace_id(trace_id_span);
+    opentelemetry::trace::SpanId span_id(span_id_span);
+    opentelemetry::trace::SpanId parent_span_id(parent_span_id_span);
+
+    rec.SetIds(trace_id, span_id, parent_span_id);
+
+    ASSERT_EQ(rec.span().trace_id(), (char*)trace_id_buff);
```

Comment:
I think there is a bug here:

rec.span().trace_id() is a string
so (char*) is implicitly converted to string
because first byte of trace_id_buff is 0 (NULL), the resulting string is empty.

It is necessary to use std::string constructor that takes size as a second argument to ensure entire buffer is copied into string, even if some symbols are null:

ASSERT_EQ(rec.span().trace_id(), std::string(reinternpret_cast<const char*>(trace_id_buff), trace::TraceId::kSize);